ports/esp8266/common-hal/pulseio/PulseOut.c

```diff
@@ -24,23 +24,41 @@
  * THE SOFTWARE.
  */
 
+#include "common-hal/pulseio/PulseOut.h"
 
 #include <stdint.h>
 
+#include <pwm.h>
+
+#include "ets_alt_task.h"
+#include "py/obj.h"
 #include "py/runtime.h"
+#include "mpconfigport.h"
 #include "shared-bindings/pulseio/PulseOut.h"
 
+void pulseout_set(pulseio_pulseout_obj_t *self, bool state) {
+    PIN_FUNC_SELECT(self->pin->peripheral, state ? self->pin->gpio_function : 0);
+}
```

Comment:
Same here with GPIO16. Will need special handling since it doesn't work with `PIN_FUNC_SELECT`.

Reply:
... and the esppwm.c implementation doesn't support GPIO16 either (only pins 0, 2, 4, 5, 12, 13, 14, 15), so you can't get a PWM object for it and thus can't construct a PulseOut object either.  There probably should be an explicit check for SPECIAL_CASE in common_hal_pulseio_pulseout_construct though, I'll add that in.

@tannewt and I discussed adding the ability to construct PulseOut around a DigitalInOut instead, which would be useful for external modulation and could be made to work on GPIO16. It's also possible we could substantially rewrite esppwm.c to work a lot better but that's beyond the scope of this PR :-)

Reply:
PS: Allowing DigitalInOut would require changes to the `shared-bindings/pulseio/PulseOut.c` so I wasn't sure that was in scope either.

Reply:
Agree that the larger change would be better outside this PR. _If_ it is desired... 😄